File: froom-backend/src/main/java/com/froom/backend/authorization/controller/AuthorizationController.java
package com.froom.backend.authorization.controller;

import com.froom.backend.authorization.model.command.AuthorizationCommand;
import com.froom.backend.authorization.model.command.RegisterUserCommand;
import com.froom.backend.authorization.model.domain.Token;
import com.froom.backend.authorization.model.dto.AuthorizationDto;
import com.froom.backend.authorization.service.AuthorizationService;
import com.froom.backend.authorization.service.JwtTokenService;
import com.froom.backend.user.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController("auth")
@RequiredArgsConstructor
public class AuthorizationController {

    private final AuthorizationService tokenService;
    private final UserService userService;

    @PostMapping("/login")
    public Token login(@RequestBody String email, @RequestBody String password) throws Exception {
        return tokenService.generateToken(email, password);
    }

    @PostMapping("/refreshToken")
    public AuthorizationDto refreshToken() {
        return new AuthorizationDto("new_token", null, null);
    }

    @PostMapping("/register")
    public ResponseEntity<String> registerUser(@RequestBody RegisterUserCommand command) {
        try {
            userService.registerUser(command);
            return ResponseEntity.ok("User registered successfully");
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Registration failed: " + e.getMessage());
        }
    }
}

File: froom-backend/src/main/kotlin/com/froom/froombackend/authorization/controller/AuthorizationController.kt
package com.froom.froombackend.authorization.controller

import com.froom.froombackend.authorization.model.command.LoginAuthCommand
import com.froom.froombackend.authorization.model.command.RefreshAuthCommand
import com.froom.froombackend.authorization.model.dto.TokenDto
import com.froom.froombackend.authorization.service.AuthorizationService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class AuthorizationController (
    private val authorizationService: AuthorizationService
) {

    @PostMapping("/login")
    fun login(@RequestBody @Valid command: LoginAuthCommand): ResponseEntity<TokenDto> {
        return ResponseEntity<TokenDto>(authorizationService.login(command),
            HttpStatus.ACCEPTED)
    }

    @PostMapping("/refresh")
    fun refresh(@RequestBody @Valid command: RefreshAuthCommand): ResponseEntity<TokenDto> {
        return ResponseEntity<TokenDto>(authorizationService.refreshToken(command),
            HttpStatus.ACCEPTED)
    }
}